source/istio_virtualservice.go

```diff
@@ -290,7 +290,10 @@ func (sc *virtualServiceSource) targetsFromVirtualService(ctx context.Context, v
 	var targets []string
 	// for each host we need to iterate through the gateways because each host might match for only one of the gateways
 	for _, gateway := range virtualService.Spec.Gateways {
-		gateway := sc.getGateway(ctx, gateway, virtualService)
+		gateway, err := sc.getGateway(ctx, gateway, virtualService)
+		if err != nil {
+			return targets, err
```

Comment:
Should this be: nil, err ?

Reply:
Not sure about this I used `return targets, err`, because in L303 it also returns the targets and the error https://github.com/kubernetes-sigs/external-dns/blob/796ebbffa8a16a0e5878eaa10ac57720fd417d96/source/istio_virtualservice.go#L303

What do you think, should I change it?

Reply:
My rule of thumb is to return nil in case you return an err.
So if there's no reason for having this in particular I would say please change.

Reply:
I adjusted it, to just return the error.